### ISSUE TITLE
Return [com_poll] as an official extension - JED

### ISSUE THREAD

**Issue by 4digitalworld** [joomla/joomla-cms#28133]
### Is your feature request related to a problem? Please describe.
Joomla `1.0` & `1.5` had a good and at the same time easy to use component for voting on the website.

This component was called `Polls` (`com_poll`).

I suggest returning this component as an official extension here - [https://extensions.joomla.org/category/official-extensions/](https://extensions.joomla.org/category/official-extensions/)

I, like many others still have backup databases, was used on the site with Joomla 1.5

I made my website in 2007 on Joomla 1.0 and now this website on @Joomla 3.9. But this component was light and simple. I do not see the same free counterparts in JED.

### Describe the solution you'd like

1. Create a directory with component `Polls` (`com_poll`) here - [https://github.com/joomla-extensions](https://github.com/joomla-extensions)
2. Make the component available for download in [JED](https://extensions.joomla.org/category/official-extensions/)

### Additional context
I see that two components [Weblinks](https://github.com/joomla-extensions/weblinks) (`com_weblinks`) and [Search](https://github.com/joomla-extensions/search) (`com_search`) were separated from the CMS and became available for download via [JED](https://extensions.joomla.org/category/official-extensions/). I propose to do the same for component `Polls` (`com_poll`).

**Joomla 1.5 Docs:** [https://docs.joomla.org/J1.5:Adding_a_new_Poll#How_to_Create_a_Poll_in_the_Poll_Manager](https://docs.joomla.org/J1.5:Adding_a_new_Poll#How_to_Create_a_Poll_in_the_Poll_Manager)

### Why?

* [x] This component was part of Joomla CMS.
* [x] He was free
* [x] It worked well and easily.
* [x] He satisfied most website editors.
* [x] Components in JED are not as light and simple and have a lot of unnecessary functionality
* [x] Surveys are still very popular on the Internet. Social networks like @Twitter and @Facebook have the function of polls in user publications.

**Comment by alikon** [comment]
are the extensions already present in the JED not enough ?
https://extensions.joomla.org/instant-search/?jed_live%5Bquery%5D=poll&jed_live%5BrefinementList%5D%5Btype%5D%5B0%5D=Free

**Comment by 4digitalworld** [comment]
> 
> 
> are the extensions already present in the JED not enough ?
> https://extensions.joomla.org/instant-search/?jed_live%5Bquery%5D=poll&jed_live%5BrefinementList%5D%5Btype%5D%5B0%5D=Free

I tried to use a couple of other extensions after my website migrated from Joomla 1.5 to Joomla 2.5 in 2012. These attempts failed. I think this component it has been undeservedly removed from the CMS. But if possible, I think it would be possible to simply return it as an official extension for the JED. It will be useful for the **legacy**. This is the only component that has been removed without the possibility of recovery.

It is also **compatible with fields in the database** if this component is returned.

This was the right decision when `Web Links` and `Search` were left as separate extensions. I constantly use `Web links`, and I started using these extensions after it was separated from the CMS. This is really a very good extension. The same thing I think about `Polls`.

**Comment by 4digitalworld** [comment]
There were a lot of polls on my website and many thousands of votes on each poll from 2007 to 2012, until I migrated to Joomla 2.5

This data is still stored in the old database. It would be cool if I could continue to use the `Polls` component. It was a very good component. Lightweight and simple. He 100% met my needs. It was as easy as Facebook polling or Twitter.

**Comment by brianteeman** [comment]
This is the first request in 7+ years - guess it really wasn't something that everyone wanted

**Comment by ReLater** [comment]
It's always the same thing: You need volunteers.

1 case taken from real life:
It wouldn't be a big problem for me to port this extension to Joomla 3 using clean Joomla-3-code and to support it **BUT** Joomla 4 is a big mistery for me (I'm a self-taught person and don't understand lots of these "strange" component changes in J4 that are "incompatible with my skills (and Joomla 3)").

Means: Even if I would do that work it would be frustrating for users and me if this component would have again an end-of-life with Joomla 4 or more work work work to make it portable then...

**Comment by brianteeman** [comment]
> It's always the same thing: You need volunteers.

In this case I would say it is more that this should not be a core feature. Nothing to do with having someone to do it

**Comment by ReLater** [comment]
> should not be a core feature

I meant more a "pseudo core feature" = continuously and reliably supported and portable ;-)

**Comment by mbabker** [comment]
When it’s proven that Weblinks is successfully maintained then additional (optional) core features can be entertained.

**Comment by 4digitalworld** [comment]
> 
> 
> When it’s proven that Weblinks is successfully maintained then additional (optional) core features can be entertained.

![изображение](https://user-images.githubusercontent.com/3662519/75611016-93d0ac80-5b27-11ea-9dfe-50ebac30d159.png)

I use `Web Links` constantly. I'm has already created more than 2000 links through this component. 

I think this is a very good component, I just realized it only a few years ago. I did not try to use this component when I used Joomla `1.0`, `1.5` and `2.5` on my website.

But I began to actively use this component when I upgraded the website to Joomla `3.x`.

It simply means that each administrator begins to use certain functions of the CMS only after some time, when he realizes the need for these functions.

I use the `Web Links` component for all external links, for affiliate links and other links. It is very comfortable. And this is very good for the SEO.

I even sometimes think, and why I have not used this component before on my website (in 2007-2012). It's great!

**Comment by 4digitalworld** [comment]
> 
> 
> This is the first request in 7+ years - guess it really wasn't something that everyone wanted

I very actively used this component as soon as I made the first website on Joomla in 2007 and used this component until I migrated to **Joomla** `2.5` in 2012.

In 2012, I did not know whom to write to save this component in the CMS. In 2012, I did not know about GitHub and talked only on Russian forums, where I was recommended to install another component with polls.

A few years ago, I found out about GitHub and saw a page here from the CMS Joomla. I saw news that components such as `Web links` and `Search` were separated from the CMS, but they would still be available in [JED](https://extensions.joomla.org/category/official-extensions/).

I think it would be great if the `Polls` component were available as a `Web links` component. It was a very good component. I did a lot of polls on my website in 2007-2012. It was a tragedy when I updated the website to Joomla 2.5 and saw that this component is no longer there.

Surveys are very popular these days on the Internet. You can see how successful it is on Twitter or Facebook. People love polls. I still do not understand why this component was removed from the CMS.

I think that many do not even know about Github, who is an active website editor using Joomla CMS. These people, just like me in 2012, could not write about their problem directly to the company that developed Joomla CMS. Many who use Joomla do not even know English at a good level, so they do not use GitHub or foreign forums (which is why forums of webmasters in local languages ​​that are spoken in different countries are so popular).

I think you can place the code on Github and if someone wants to, then can make this component compatible with Joomla 3 & Joomla 4

You can just post it on Github and the rest can be done by developers who need this component.

**Comment by mbabker** [comment]
You clearly missed the sarcasm.  All code that is not part of this repository is effectively either monitored by one person or not being looked after consistently.  Joomla doesn’t have the resources to make more optional features.  Hell it barely has the resources to keep this repo running.

The code is GPL.  Anyone is free to extract the 1.5 component out of that package and publish it on their own.  But please, Joomla, stop spending Human Resources on new endeavors you can’t support.

**Comment by 4digitalworld** [comment]
> 
> 
> You clearly missed the sarcasm. All code that is not part of this repository is effectively either monitored by one person or not being looked after consistently. Joomla doesn’t have the resources to make more optional features. Hell it barely has the resources to keep this repo running.
> 
> The code is GPL. Anyone is free to extract the 1.5 component out of that package and publish it on their own. But please, Joomla, stop spending Human Resources on new endeavors you can’t support.

There is no sarcasm. You say as if I asked to create a system with artificial intelligence. This is simply a request for porting an already created component that was part of the CMS. Nothing incredible.

This has already been created for the CMS. Just put this on Github as an official extension. Nothing more.

I think there will be people who want to use this component again. This is very much in demand, also in demand, like polls on social networks. You can even see what `Polls` are doing on @Instagram **Stories** right now. Wow!

**Comment by mbabker** [comment]
> Nothing incredible

Then you clearly don’t understand the commitment it takes to build and maintain a system such as that.  Including the effort it would take to get a releasable product.  This part alone is easily a $10,000 project, I don’t know of many people that would willingly volunteer that much of their time on a project of this scope and assume all the financial risks working on something that has no demand, especially when there are already several dozen projects within Joomla which already have equal or greater resource requirements.

If someone wants to release core’s poll extension from 1.5 on their own accord, feel free to do so.  But, once again, I can say with confidence that Joomla doesn’t have the resources to do this and the project shouldn’t entertain the idea.

**Comment by 4digitalworld** [comment]
> 
> 
> > Nothing incredible
> 
> Then you clearly don’t understand the commitment it takes to build and maintain a system such as that. Including the effort it would take to get a releasable product. This part alone is easily a $10,000 project, I don’t know of many people that would willingly volunteer that much of their time on a project of this scope and assume all the financial risks working on something that has no demand, especially when there are already several dozen projects within Joomla which already have equal or greater resource requirements.
> 
> If someone wants to release core’s poll extension from 1.5 on their own accord, feel free to do so. But, once again, I can say with confidence that Joomla doesn’t have the resources to do this and the project shouldn’t entertain the idea.

You can place the code for this component, which was for Joomla 1.5 here - [https://github.com/joomla-extensions](https://github.com/joomla-extensions) and write about compatibility with Joomla 1.5

Also, when the news comes out about the next version of Joomla `3.9.16` on the website [https://joomla.org](https://joomla.org), write something like this:

> Component `Polls` (`com_poll`) is available on our [Github](https://github.com/joomla-extensions). If you want to take part in the development of this component, write to us. You can also join the development of component [Web links](https://github.com/joomla-extensions/weblinks).

I think you can find volunteers who want to develop this component. You just don’t understand that there are many people who would like to write a line about participating in Open Source in their portfolio.

**Comment by Bakual** [comment]
I'm closing this request. com_poll was removed for a reason.
Also there are enough (better) alternatives available on JED.
There is no reason to have this in core.

**Comment by 4digitalworld** [comment]
> 
> 
> > Nothing incredible
> 
> $10,000 project

![изображение](https://user-images.githubusercontent.com/3662519/75613337-655dcc00-5b3d-11ea-95ec-65377a778032.png)

I think that additional money for the development of @Joomla can be obtained if you specify a link to this website https://opencollective.com (@opencollective) - as the @twbs team did.

I mean, specify the link to [https://opencollective.com](https://opencollective.com) from the site [https://joomla.org](https://joomla.org)
